flood-fill.py

```diff
@@ -1,14 +1,89 @@
+from pprint import pprint
 grid = [
         ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R'],
-        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R']
-        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R']
-        ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R']
+        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'R', 'R', 'R', 'Y', 'R', 'R'],
+        ['R', 'R', 'Y', 'Y', 'Y', 'Y', 'Y', 'R', 'R'],
+        ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R'],
         ['R', 'R', 'R', 'R', 'R', 'R', 'R', 'R', 'R']
 ]
 
+
 def fill(row, col, new_color):
-    return # TODO practice for geoff
\ No newline at end of file
+        global grid
```

Comment:
don't think you need to use global here because you only mutate the values in the array and not the array itself

Reply:
I tested it without the globals and you were right. I don't understand the difference between mutating values and mutating the array itself though.

Reply:
mutating the array would be like:

grid = some new array

mutating values is:
grid[row][col] = bla